File: backend/internal/service/talento_service.go
package service

import "rpg-manager/internal/domain"

type talentoRepo interface {
	GetAll(edition string) ([]domain.Talento, error)
	GetByCharacter(characterID uint) ([]domain.Talento, error)
	Add(characterID, talentoID uint) error
	Remove(characterID, talentoID uint) error
}

type TalentoService struct{ repo talentoRepo }

func NewTalentoService(repo talentoRepo) *TalentoService { return &TalentoService{repo: repo} }

func (s *TalentoService) GetAll(edition string) ([]domain.Talento, error) {
	return s.repo.GetAll(edition)
}
func (s *TalentoService) GetByCharacter(id uint) ([]domain.Talento, error) {
	return s.repo.GetByCharacter(id)
}
func (s *TalentoService) Add(charID, talentoID uint) error {
	return s.repo.Add(charID, talentoID)
}
func (s *TalentoService) Remove(charID, talentoID uint) error {
	return s.repo.Remove(charID, talentoID)
}

File: frontend/src/services/talentoService.ts
import api from './api'
import type { Talento } from '../types'

export const talentoService = {
  getAll: async (edition?: string): Promise<Talento[]> => {
    const { data } = await api.get('/talentos', {
      params: edition ? { edition } : {},
    })
    return data
  },

  getByCharacter: async (characterId: number): Promise<Talento[]> => {
    const { data } = await api.get(`/characters/${characterId}/talentos`)
    return data
  },

  add: async (characterId: number, talentoId: number): Promise<void> => {
    await api.post(`/characters/${characterId}/talentos/${talentoId}`)
  },

  remove: async (characterId: number, talentoId: number): Promise<void> => {
    await api.delete(`/characters/${characterId}/talentos/${talentoId}`)
  },
}